ci: Add job to ensure python2 setup.py compat

seemethere opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#56057 ci: Add job to ensure python2 setup.py compat**

Just to make sure we don't add anything there that'd break python 2 users from receiving the correct error message

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D27774120](https://our.internmc.facebook.com/intern/diff/D27774120)

seemethere committed:
ci: Add workflow to ensure python2 setup.py compat

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere committed:
Update on "ci: Add workflow to ensure python2 setup.py compat"


Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere committed:
Update on "ci: Add workflow to ensure python2 setup.py compat"


Just to make sure we don't add anything there that'd break python 2 users from receiving the correct error message

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere committed:
Update on "ci: Add workflow to ensure python2 setup.py compat"


Just to make sure we don't add anything there that'd break python 2 users from receiving the correct error message

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere committed:
Update on "ci: Add workflow to ensure python2 setup.py compat"


Just to make sure we don't add anything there that'd break python 2 users from receiving the correct error message

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

malfet reviewed (approved):
LGTM, but can you test it before landing (for example by  changing message to "Python 3 has reached end-of-life"?)
I though of `set -x` but it would fail earlier, as `python setup.py` returns -1

malfet commented on .github/workflows/lint.yml:
```diff
@@ -69,6 +69,20 @@ jobs:
+        uses: actions/checkout@v2
+      - name: Attempt to run setup.py
+        run: |
+          python setup.py | grep "Python 2 has reached end-of-life and is no longer supported by PyTorch."
```
Nit (python alias can point to both python2 and python3, but python2 guaranteed to point to python-2
```suggestion
          python2 setup.py | grep "Python 2 has reached end-of-life and is no longer supported by PyTorch."
```

seemethere commented:
> LGTM, but can you test it before landing (for example by changing message to "Python 3 has reached end-of-life"?)
> I though of `set -x` but it would fail earlier, as `python setup.py` returns -1

Should work as expected:
```
❯ docker run --rm -it -v $(pwd):/v -w /v python:2 python setup.py |  grep "Python 2 has reached end-of-life and is no longer supported by PyTorch." 

❯ echo $?
1
```

seemethere committed:
Update on "ci: Add job to ensure python2 setup.py compat"


Just to make sure we don't add anything there that'd break python 2 users from receiving the correct error message

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D27774120](https://our.internmc.facebook.com/intern/diff/D27774120)

[ghstack-poisoned]
